geonotebook/vis/ktile/ktile.py

```diff
@@ -201,20 +298,24 @@ def ingest(self, data, name=None, **kwargs):
             options.update(self._dynamic_vrt_options(data, kwargs))
 
         # Make the Request
-        base_url = '{}/{}/{}'.format(self.base_url, kernel_id, name)
-
-        r = requests.post(base_url, json={
-            "provider": {
-                "class": "geonotebook.vis.ktile.provider:MapnikPythonProvider",
-                "kwargs": options
+        port_request = self.webapp_comm({ 'action': 'request_port' })
+        base_url = 'http://127.0.0.1:{}/ktile/{}/{}'.format(port_request['port'], kernel_id, name)
```

Comment:
In fact, `webapp.settings['base_url'] == '/'` and seems to have no knowledge of the host.  Localhost seems to be the best default, but I'm sure there are scenarios where this will fail.  I'll entertain other suggestions on this point if you have them.

Reply:
So, scratch that.  The webapp-provided base URL is fine.  Eliminates the need for the port at all, and makes the extension agnostic to whether it is being run under JupyterHub or vanilla Jupyter.